pkg/bgpv1/agent/controller.go

```diff
@@ -134,6 +145,20 @@ func (c *Controller) Start(startCtx hive.HookContext) error {
 		return nil
 	})
 
+	if c.SecretResource != nil {
+		c.workerpool.Submit("secret-observer", func(ctx context.Context) error {
+			for ev := range c.SecretResource.Events(ctx) {
+				switch ev.Kind {
+				case resource.Upsert:
+					// Signal the reconciliation logic.
+					c.Sig.Event(struct{}{})
+				}
+				ev.Done(nil)
```

Comment:
The idea was that for manually managed secrets a common workflow is to `kubectl delete` and then `kubectl create`, which would actually bring down a BGP session and then bring it up.

But this touches on your other comment too about handling missing secrets. Having some sort of retry for fetching secrets which doesn't change the state (i.e. eventual reconciliation, much like how Pods work) would solve both. It's a bit hard to do with the current structure as it assumes everything is driven off policy changes, but I'll have a think.

Reply:
Sorry, maybe I'm not following. Why can't we just update the secret instead of delete & create? I though if we see the delete event, the right behavior would be do reconcile and shutdown the BGP session. Am I missing something?

Reply:
If someone is managing the secret with kubectl directly, then it is a surprisingly complex command to update the secret (see https://stackoverflow.com/questions/45879498/how-can-i-update-a-secret-on-kubernetes-when-it-is-generated-from-a-file) -- I suspect maybe this isn't that common for people using BGP, as BGP is the kind of thing that needs a lot of infrastructure, which to me implies some kind of CD / secrets management system.

But it's also the kind of place where personally if I was dealing with a broken cluster and updating a secret was needed, it would be surprising to me that deleting a secret tears down a BGP session -- secrets or configmaps going away doesn't delete a pod.

I've adjusted the handling now so it handles delete, but if there was previously a valid secret it uses it.

Reply:
Ok, I understand. For me, it's ok to go with this behavior. Users still can "delete" a secret by deleting `authSecretRef` and the result is the same as handling the delete event.